inc/az_ulib_ipc_api.h

```diff
@@ -310,31 +282,20 @@ static inline az_result az_ulib_ipc_release_interface(az_ulib_ipc_interface_hand
  *                                input model content.
  * @param[out]  model_out         The `const void *` that points to the memory where the capability
  *                                should store the output model content.
+ *
+ * @pre     IPC shall already been initialized.
+ * @pre     \p interface_handle shall be different than `NULL`.
+ *
  * @return The #az_result with the result of the call.
  *  @retval #AZ_OK                              If the IPC get success calling the procedure.
- *  @retval #AZ_ERROR_ARG                       If one of the arguments is invalid.
  *  @retval #AZ_ERROR_ITEM_NOT_FOUND            If the target command was disabled.
- *  @retval #AZ_ERROR_CANCELED                  If the target command was unpublished.
- *  @retval #AZ_ERROR_ULIB_BUSY                 If the target command cannot be executed at that
- *                                              moment.
  */
-static inline az_result az_ulib_ipc_call(
+AZ_NODISCARD az_result az_ulib_ipc_call(
```

Comment:
Hey Marcos, just for my education, did you get smaller image size by not making these inline functions? What was the main motivation for the change?

Reply:
This inline was only to decide between "Precondition" enable or disable that we are doing now using az_PRECONDITION macros. So, this inline was not helping in size or performance. For the final implementation, we will probably consider not only inline, but bypass the C arguments code convention, so we will avoid to store all args twice in the stack. But it will be part of another PR.

---
In reply to: [590630423](https://github.com/Azure/azure-ulib-c/pull/61#discussion_r590630423) [](ancestors = 590630423)